### PR TITLE
feat(ci): Add new methods to generate full gitlab configuration

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Add a new method (not called yet) to generate a gitlab-ci full configuration by concatenation of all the `include` recursively

### Motivation
This step is required to be able to use the `lint` [API endpoint](https://docs.gitlab.com/ee/api/lint.html) of gitlab. This will improve the possible tests we do when we modify the ci yaml configuration files

### CODE REVIEW

**Review by ddyurchenko — APPROVED** [review]

**Comment by chouquette** [comment]
Hello :wave: 
I'd split the yml files change in a dedicated PR which will likely be easily mergeable, and which will reduce the number of reviewers being pulled it for the actual new task

**Comment by chouetz** [comment]
> Hello 👋 I'd split the `./gitlab/**/*.yml` files change in a dedicated PR which will likely be easily mergeable, and which will reduce the number of reviewers being pulled it for the actual new task

ok why not :D

**Review by richl-dd — APPROVED** [review]

**Review by ahmed-mez — APPROVED** [review]

**Review by lrbhatia — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

File: tasks/linter.py
[the remark this answers is not in the document]
Comment (chouetz) [reply tasks/linter.py:1]: Done

**Pull request merged**